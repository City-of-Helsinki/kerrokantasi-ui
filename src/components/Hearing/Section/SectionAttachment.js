import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../../../utils/Icon';


/**
 * Determines the correct title to return
 * @param {Object} title - title object
 * @param {String} language - language of application
 */
const getFileTitle = (title, language) => {
  if (title && title[language] && typeof title[language] === 'string') {
    return title[language];
  }
  return title[Object.keys(title)];
};

/**
 * Class declaration
 * Renders list of attachments available to a section.
 */
const SectionAttachment = (props) => (
  <div className="section-attachment">
    <Icon className="icon" name="file"/>
    <a href={props.file.url} className="section-attachment-title" target="_blank" rel="noreferrer">
      {getFileTitle(props.file.title, props.language)}
    </a>
  </div>
);

SectionAttachment.propTypes = {
  file: PropTypes.object.isRequired,
  language: PropTypes.string.isRequired,
};

export default SectionAttachment;
export { getFileTitle };
