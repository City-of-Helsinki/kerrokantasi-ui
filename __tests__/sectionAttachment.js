import React from 'react';
import {shallow, mount} from 'enzyme';
import SectionAttachment, { getFileTitle } from '../src/components/Hearing/Section/SectionAttachment';

const props = {
  language: 'fi',
  file: { 
    url: 'http://mock.url',
    title: {
      fi: 'Title in finnish',
      en: 'Title in english',
    }
  }
};

let wrapper;

describe('<SectionAttachment />', () => {
  beforeEach(() => {
    wrapper = shallow(<SectionAttachment { ...props } />);
  });

  test('Should match snapshot' , () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render the component', () => {
    expect(wrapper.find('.section-attachment').length).toBe(1);
  });

  test('Should get the right file name based on language', () => {
    const title = getFileTitle(props.file.title, props.language);
    expect(title).toBe(props.file.title.fi);
  });

  test('Should get English language title when language is english', () => {
    const title = getFileTitle(props.file.title, 'en');
    expect(title).toBe(props.file.title.en);
  });
});
