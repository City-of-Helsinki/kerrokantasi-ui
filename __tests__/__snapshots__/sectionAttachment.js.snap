// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionAttachment /> Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SectionAttachment
    file={
      Object {
        "title": Object {
          "en": "Title in english",
          "fi": "Title in finnish",
        },
        "url": "http://mock.url",
      }
    }
    language="fi"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          Component="span"
          className="icon"
          name="file"
        />,
        <a
          className="section-attachment-title"
          href="http://mock.url"
          rel="noreferrer"
          target="_blank"
        >
          Title in finnish
        </a>,
      ],
      "className": "section-attachment",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "Component": "span",
          "className": "icon",
          "name": "file",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title in finnish",
          "className": "section-attachment-title",
          "href": "http://mock.url",
          "rel": "noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": "Title in finnish",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            Component="span"
            className="icon"
            name="file"
          />,
          <a
            className="section-attachment-title"
            href="http://mock.url"
            rel="noreferrer"
            target="_blank"
          >
            Title in finnish
          </a>,
        ],
        "className": "section-attachment",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "Component": "span",
            "className": "icon",
            "name": "file",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title in finnish",
            "className": "section-attachment-title",
            "href": "http://mock.url",
            "rel": "noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Title in finnish",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
